### PR TITLE
fix(ground_filter): expand range of cropped_box_filter

### PR DESCRIPTION
**Pull request by badai-nguyen**
Signed-off-by: badai-nguyen <dai.nguyen@tier4.jp>

## Description

Current scan_ground_filter use the input from cropped_box_filter which limited from 0 < z < vehicle_height. Therefore, in case of uneven road or slope, the ground pcl might be filtered out and failure to initialize initial point cloud by scan_ground_filter. It could fixed by extending the cropped_box_filter range wider. This makes sure scan_ground_filter excludes the under ground point from non_ground point label when extending the range of cropped_box_filter. 

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: perception/ground_segmentation/src/scan_ground_filter_nodelet.cpp [hunk 1]
```diff
@@ -168,7 +170,13 @@ void ScanGroundFilterComponent::classifyPointCloud(
 
       float global_slope = std::atan2(p->orig_point->z, p->radius);
       // check points which is far enough from previous point
-      if (global_slope > global_slope_max_angle) {
+      if (
+        (height_from_gnd < -split_height_distance_) ||
+        ((p->orig_point->z - ground_cluster.getAverageHeight()) < -split_height_distance_) ||
+        (height_from_gnd > non_ground_heigh_thresh_)) {
+        // exclude pcl under ground or higher than limit
+        calculate_slope = false;
```
Comment (miursh) [inline perception/ground_segmentation/src/scan_ground_filter_nodelet.cpp:178]: In this case, what would be the label for point?

Comment (badai-nguyen) [reply perception/ground_segmentation/src/scan_ground_filter_nodelet.cpp:181]: I assigned as out_range label for that points

File: perception/ground_segmentation/src/scan_ground_filter_nodelet.cpp [hunk 2]
```diff
@@ -168,7 +170,13 @@ void ScanGroundFilterComponent::classifyPointCloud(
 
       float global_slope = std::atan2(p->orig_point->z, p->radius);
       // check points which is far enough from previous point
-      if (global_slope > global_slope_max_angle) {
+      if (
+        (height_from_gnd < -split_height_distance_) ||
```
[the remark this answers is not in the document]
Comment (badai-nguyen) [reply perception/ground_segmentation/src/scan_ground_filter_nodelet.cpp]: thanks for your comment, I fixed.

File: perception/ground_segmentation/src/scan_ground_filter_nodelet.cpp (hunk 1 above)
Comment (miursh) [reply perception/ground_segmentation/src/scan_ground_filter_nodelet.cpp:183]: When the point is labeled out of range, prev point label [will be also set](https://github.com/badai-nguyen/autoware.universe/blob/fix/ground_filter/perception/ground_segmentation/src/scan_ground_filter_nodelet.cpp#L231) to out of range. Is this okay for next point judgement?

**Comment by miursh** [comment]
@badai-nguyen Could you update readme about the new parameters?

**Comment by miursh** [comment]
I think you should update the crop box range [here](https://github.com/badai-nguyen/autoware.universe/blob/fix/ground_filter/launch/tier4_perception_launch/launch/obstacle_segmentation/ground_segmentation/ground_segmentation.launch.py#L60)

File: perception/ground_segmentation/src/scan_ground_filter_nodelet.cpp (hunk 1 above)
Comment (badai-nguyen) [reply perception/ground_segmentation/src/scan_ground_filter_nodelet.cpp:183]: as [this condition ](https://github.com/badai-nguyen/autoware.universe/blob/c786f6b5c62fe488430222b64394cd12fe6492b5/perception/ground_segmentation/src/scan_ground_filter_nodelet.cpp#L191) , if the point is labeled out of range, the next point should not classify as POINT_FOLLOW, but GROUND, NON_GROUND or OUT_OF_RANGE. 
Keeping to use the current ground_cluster and non_ground_cluster to check the next point is reasonable, IMO.

**Comment by miursh** [comment]
@badai-nguyen 
It still can not able to detect the vehicle on the other side of the intersection
Screenshot from 2022-07-07 16-29-22
What happens if we extend the range further down? Will there be too many false positives?

**Comment by miursh** [comment]
@badai-nguyen  Detection performance seems to be worse than before in some areas. For example the vehicle approaching from left side to the intersection looks like this.

[left: before,right: after]

image

image

**Comment by badai-nguyen** [comment]
By extending more range of crop_box, the vehicle on the other side of intersection can be detected at few moments. Not perfect yet.
Screenshot from 2022-07-08 00-10-07

**Comment by badai-nguyen** [comment]
A more general approach is needed to address this issue.

**Pull request closed without merging**